### PR TITLE
Add support for the GismoConfig format found in Sonic Colors and Sonic Lost World

### PR DESCRIPTION
**Pull request by DeaTh-G**
Adds read-write support for both the `*.orc (Sonic Colors)` and `*.gism (Sonic Lost World)` gismo config files.

### CODE REVIEW

**Review by Sajidur78 — CHANGES_REQUESTED** [review]

File: Source/SharpNeedle/SonicTeam/GismoConfig.cs [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+                writer.Write(Convert.ToInt32(IsMotionOn));
+                writer.WriteObjectOffset(Motion);
+
+                writer.Write(Convert.ToInt32(IsProgramMotionOn));
+                writer.WriteObjectOffset(ProgramMotion);
+            }
+        }
+
+        public enum EShapeType : int
```
[the remark this answers is not in the document]
Comment (DeaTh-G) [reply Source/SharpNeedle/SonicTeam/GismoConfig.cs]: Resolved in ab3d90b.

File: Source/SharpNeedle/SonicTeam/GismoConfig.cs [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+            AnimationName = reader.ReadStringOffset();
+        }
+
+        public void Write(BinaryObjectWriter writer)
+        {
+            writer.Write(PlayPolicy);
+            writer.WriteStringOffset(StringBinaryFormat.NullTerminated, AnimationName);
+        }
+
+        public enum EPlayPolicy : int
```
[the remark this answers is not in the document]
Comment (DeaTh-G) [reply Source/SharpNeedle/SonicTeam/GismoConfig.cs]: Resolved in ab3d90b.

File: Source/SharpNeedle/SonicTeam/GismoConfig.cs [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+            writer.Write(MotionType);
+
+            writer.Write(Axis);
+
+            writer.Write(Power);
+            writer.Write(SpeedScale);
+            writer.Write(Time);
+        }
+
+        public enum EMotionType : int
```
[the remark this answers is not in the document]
Comment (DeaTh-G) [reply Source/SharpNeedle/SonicTeam/GismoConfig.cs]: Resolved in ab3d90b.

File: Source/SharpNeedle/SonicTeam/GismoConfig.cs [hunk 4]
```diff
@@ -0,0 +1,248 @@
+﻿using SharpNeedle.BINA;
+
+namespace SharpNeedle.SonicTeam;
```
Comment (Sajidur78) [inline Source/SharpNeedle/SonicTeam/GismoConfig.cs:3]: Namespace declaration should be at the top.

**Review by Sajidur78 — CHANGES_REQUESTED** [review]

File: Source/SharpNeedle/SonicTeam/GismoConfig.cs (hunk 4 above)
Comment (DeaTh-G) [reply Source/SharpNeedle/SonicTeam/GismoConfig.cs]: Resolved in 17466e9.

**Review by Sajidur78 — APPROVED** [review]
LGTM

**Pull request merged**